### DISCUSSION TITLE
MetadataPostProcess Processes Entire Directory For Each Track

### DISCUSSION THREAD

**Discussion by misleadingrhino** [RandomNinjaAtk/docker-lidarr-extended#204] [Q&A] [answered]
As best I can tell, the MetadataPostProcess.sh script seems to run through the entire album directory when it's called, but it looks like it's being called for each track when it's retagged. Is this intentional?

https://gist.github.com/misleadingrhino/6bc96e0ecfd76efafa0dd9eb8f3177db

**Comment by RandomNinjaAtk** [comment, marked as answer]
You should turn off the trigger for tagging, only set it to:
![image](https://user-images.githubusercontent.com/8386416/219700119-fd3a5594-f5c2-46b0-9832-364638f43e3d.png)

It is intentional to process every track, but because you have it triggered on tagging event, its processing multiple times, instead of just one single time.

**Reply by misleadingrhino** [reply]
Ok, thanks! When I used the AutoConfig, the trigger for the MetadataPostProcess.sh was set to "On Retag." Is that the default?
